lib/parallel_tests/grouper.rb

```diff
@@ -28,7 +28,31 @@ def in_even_groups_by_size(items, num_groups, options = {})
           raise 'Number of isolated processes must be less than total the number of processes'
         end
 
-        if isolate_count >= 1
+        if options[:specify_groups]
+          specify_spec_processes = options[:specify_groups].split('|')
+          specified_specs = options[:specify_groups].split(/[,|]/)
+          if specify_spec_processes.count > num_groups
+            raise 'Number of processes separated by pipe must be less than or equal to the total number of processes'
+          end
+
+          specified_items, items = items.partition do |item, _size|
+            specified_specs.any? { |pattern| item =~ /#{pattern}/ }
+          end
+
+          if (specified_specs - specified_items.map(&:first)).any?
+            raise 'Could not find all specs from --specify-spec-processes in main selected files & folders'
```

Comment:
should say which to make it easier to debug

Reply:
ok will add